internal/provider/cql_auth_data_source.go

```diff
@@ -43,20 +57,82 @@ func DataSourceCQLAuth() *schema.Resource {
 	}
 }
 
-func dataSourceCQLAuthRead(d *schema.ResourceData, meta interface{}) error {
+func dataSourceCQLAuthRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	var (
-		c         = meta.(*scylla.Client)
-		clusterID = d.Get("cluster_id").(int)
+		c          = meta.(*scylla.Client)
+		clusterID  = int64(d.Get("cluster_id").(int))
+		dcID, dcOK = d.GetOk("datacenter_id")
+		dns        = d.Get("dns").(bool)
 	)
 
-	conn, err := c.Connect(int64(clusterID))
+	conn, err := c.Connect(clusterID)
 	if err != nil {
-		return fmt.Errorf("error reading connection details: %w", err)
+		return diag.Errorf("error reading connection details: %w", err)
+	}
+
+	if len(conn.Datacenters) == 0 {
+		return diag.Errorf("error reading datacenter connections: not found")
+	}
+
+	var dc *model.DatacenterConnection
+
+	if dcOK {
+		datacenters, err := c.ListDataCenters(clusterID)
+		if err != nil {
+			return diag.Errorf("error reading datacenters: %w", err)
+		}
+
+	lookup:
+		for i := range datacenters {
+			lhs := &datacenters[i]
+
+			if lhs.ID == int64(dcID.(int)) {
+				for j := range conn.Datacenters {
+					rhs := &conn.Datacenters[j]
+
+					if strings.EqualFold(lhs.Name, rhs.Name) {
+						dc = rhs
+						break lookup
+					}
+				}
+			}
+		}
+
+		if dc == nil {
+			return diag.Errorf("error looking up datacenter: not found")
+		}
+	} else {
+		dc = &conn.Datacenters[0]
+	}
+
+	var seeds string
+
+	if dns {
+		if len(dc.DNS) == 0 {
+			return diag.Errorf("error reading %q datacenter dns: not found", dc.Name)
+		}
+
+		seeds = strings.Join(dc.DNS, ",")
```

Comment:
shouldn't this logic be in the api itself? I imagine frontend and other also need to know what is a seed without trying to figure it out

Reply:
[Right](https://github.com/scylladb/siren-frontend/issues/2958#issuecomment-1406418684) 😅